src/basis/clr/PrimUtils_.sml

```diff
@@ -478,9 +488,16 @@ fun fromInt i = Prim.toWord64(Prim.I42I8(i))
 
 (*@TODO: +,-,* need to avoid overflow checks and test! *)
 (*@BUG: +,-,* may raise overflow *)
-fun x + y = Prim.toWord64 (Prim.add_ovf_un(Prim.fromWord64 x, Prim.fromWord64 y))
+(* fun x + y = Prim.toWord64 (Prim.add_ovf_un(Prim.fromWord64 x, Prim.fromWord64 y))
 fun x - y = Prim.toWord64 (Prim.sub_ovf_un(Prim.fromWord64 x, Prim.fromWord64 y))
-fun x * y = Prim.toWord64 (Prim.mul_ovf_un(Prim.fromWord64 x, Prim.fromWord64 y))
+fun x * y = Prim.toWord64 (Prim.mul_ovf_un(Prim.fromWord64 x, Prim.fromWord64 y)) *)
+(* TODO: Check if this is correct or not. This looks similar to what C# does
+   in that it uses the add, sub, and mul for unchecked operations between ulongs,
+   but all the conversions between ints and words look suspicious.
+ *)
+fun x + y = Prim.toWord64 (Prim.add(Prim.fromWord64 x, Prim.fromWord64 y))
+fun x - y = Prim.toWord64 (Prim.sub(Prim.fromWord64 x, Prim.fromWord64 y))
+fun x * y = Prim.toWord64 (Prim.mul(Prim.fromWord64 x, Prim.fromWord64 y))

```

Comment:
`Word64` operations don't check for overflow, using the base `add`, `sub`, and `mul` instructions